include/clink/feature/one_hot_encoder.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_FEATURE_ONE_HOT_ENCODER_H_
+#define CLINK_FEATURE_ONE_HOT_ENCODER_H_
+
+#include "clink/feature/proto/one_hot_encoder.pb.h"
+#include "clink/linalg/sparse_vector.h"
+#include "llvm/Support/Errc.h"
+#include "llvm/Support/Error.h"
+
+namespace clink {
+
+// A Model which encodes data into one-hot format.
+class OneHotEncoderModel {
+public:
+  // Default constructor.
+  OneHotEncoderModel() {}
+
+  // Move operations are supported.
+  OneHotEncoderModel(OneHotEncoderModel &&other) = default;
+  OneHotEncoderModel &operator=(OneHotEncoderModel &&other) = default;
+
+  // This class is copyable and assignable.
+  OneHotEncoderModel(const OneHotEncoderModel &other) = default;
+  OneHotEncoderModel &operator=(const OneHotEncoderModel &) = default;
+
+  // Converts the provided data to a SparseVector.
+  llvm::Expected<SparseVector> transform(const int value,
+                                         const int columnIndex);
+
+  // Loads a OneHotEncoderModel from given path.
+  static llvm::Expected<OneHotEncoderModel> load(const std::string path);
```

Comment:
Could you explain what is the contract of this load method? How is user supposed to use this load(path) method?

Note that the contract of the `load(path)` in the Java stage class is that we can re-build exactly the same stage that was saved using the `save(path)`, which is well-defined and self-contained. But we don't have `save(...)` on this C++ OneHotEncoderModel class.

Reply:
This `load(path)` method should load data that was saved by `ClinkOneHotEncoderModel.save()`. C++ `OneHotEncoderModel` does not have `save(...)` method because in our expected use case we don't need to train and save model in C++ environment.

The usage of these APIs are as follows as I can conceive:
- Creates `ClinkOneHotEncoder` in Java
- Gets `ClinkOneHotEncoderModel` with the `ClinkOneHotEncoder` and train data in Java
- Saves `ClinkOneHotEncoderModel`'s params and model data in a directory in Java
- Loads the saved data and creates `OneHotEncoderModel` in C++
- Uses the `OneHotEncoderModel` to do inference in mlir scripts or other C++ frameworks

Reply:
Just to double check, we could first do `clinkOneHotEncoderModel::save("/tmp/mydir")` and then do `llvm::Expected<OneHotEncoderModel> model = load("/tmp/mydir")`. This is correct?

Reply:
Yes. That's correct.

Reply:
As discussed offline, Flink cannot save to a certain file and only allows to save to a directory. Thus in C++ code we can only look through the designated directory for possible model data files and load them.